### PR TITLE
Migrate from filePaths to URIs with latest Cody

### PR DESCRIPTION
**Pull request by Strum355**
See https://github.com/sourcegraph/cody/commit/2b3c3b0a708d11f1f828130c37116315ed140468

## Test plan

Locally tested as part of daily development

### CODE REVIEW

File: src/main/kotlin/com/sourcegraph/cody/agent/protocol/AutocompleteParams.kt [hunk 1]
```diff
@@ -1,13 +1,22 @@
...
     val triggerKind: String? = AutocompleteTriggerKind.AUTOMATIC.value,
     val selectedCompletionInfo: SelectedCompletionInfo? = null
-)
+) {
+  constructor(
+      filePath: String,
+      position: Position,
+      triggerKind: String? = AutocompleteTriggerKind.AUTOMATIC.value,
+      selectedCompletionInfo: SelectedCompletionInfo? = null
+  ) : this(URI("file://$filePath"), position, triggerKind, selectedCompletionInfo)
```
Comment (olafurpg) [inline src/main/kotlin/com/sourcegraph/cody/agent/protocol/AutocompleteParams.kt]: This URI constructor is going to crash in windows. It's better to go via nio.file.Path

Comment (Strum355) [reply src/main/kotlin/com/sourcegraph/cody/agent/protocol/AutocompleteParams.kt]: Do you have a source on this? The only historical info I can find on windows path issues was on the Cody side of things and I couldnt find anything on google

Comment (olafurpg) [reply src/main/kotlin/com/sourcegraph/cody/agent/protocol/AutocompleteParams.kt]: We already had the same issue before, which got fixed here https://sourcegraph.com/github.com/sourcegraph/jetbrains/-/commit/ac233c5959d5aff51ca4148a8d7cc5f883ac3581?visible=2

Comment (olafurpg) [reply src/main/kotlin/com/sourcegraph/cody/agent/protocol/AutocompleteParams.kt]: The problem was space characters based on that commit

Comment (Strum355) [reply src/main/kotlin/com/sourcegraph/cody/agent/protocol/AutocompleteParams.kt]: Ah yes I remember now! Couldn't find it in slack or gh issue/PR search 🤔

Comment (olafurpg) [reply src/main/kotlin/com/sourcegraph/cody/agent/protocol/AutocompleteParams.kt]: `type:diff` in Sourcegraph found 😜

Comment (Strum355) [reply src/main/kotlin/com/sourcegraph/cody/agent/protocol/AutocompleteParams.kt]: Done https://github.com/sourcegraph/jetbrains/pull/145/commits/9e2540ca0c88d35768e508a319d7f11f0432dfee

**Review by olafurpg — APPROVED** [review]

**Pull request merged**